### PR TITLE
[freetype] add subpixel-rendering feature

### PR DESCRIPTION
**Pull request by abique**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

This PR adds the feature `subpixel-rendering` to freetype.

### CODE REVIEW

**Comment by Neumann-A** [comment]
This breaks --editable workflows.

**Comment by Neumann-A** [comment]
Since this is a CMake based port using `#cmakedefine` with `configure_file` might be a better approach

**Comment by abique** [comment]
> Since this is a CMake based port using `#cmakedefine` with `configure_file` might be a better approach

I didn't know about `configure_file`. I'll take a look, thanks!

**Comment by abique** [comment]
@Neumann-A after a first look at `configure_file` I don't think it'll do the job, because it replaces `#cmakedefine XXX` or this isn't what the source file contains.
Also I don't understand what you mean with `--editable` workflow.
You can see that archlinux also uses a patch approach when packaging freetype.

**Comment by Neumann-A** [comment]
Then move the patch into the download function instead of applying it later.

**Comment by abique** [comment]
> Then move the patch into the download function instead of applying it later.

How to apply this patch conditionally then?

**Comment by Neumann-A** [comment]
> How to apply this patch conditionally then?

Use variable expansion? `PATHES ${some_var}` and set `some_var `conditionally?

**Comment by abique** [comment]
It seems that the buildbot is stuck.

**Comment by abique** [comment]
!rebuild

**Comment by abique** [comment]
Even if the buildbot doesn't go through, I think it is safe to merge.

**Comment by FrankXie05** [comment]
Re-open this PR for CI.

**Comment by abique** [comment]
Ah build went through! :+1: 
Time to merge :v:

**Review by FrankXie05 — APPROVED** [review]

File: ports/freetype/subpixel-rendering.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/include/freetype/config/ftoption.h b/include/freetype/config/ftoption.h
+index 1976b33af959..b3425e55feec 100644
+--- a/include/freetype/config/ftoption.h
++++ b/include/freetype/config/ftoption.h
+@@ -123,7 +123,7 @@ FT_BEGIN_HEADER
+    * When this macro is not defined, FreeType offers alternative LCD
+    * rendering technology that produces excellent output.
+    */
+-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
++#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
```
Comment (BillyONeal) [inline ports/freetype/subpixel-rendering.patch:10]: Confirming that they actually say to edit the header to do this 🤯

https://freetype.org/freetype2/docs/reference/ft2-lcd_rendering.html#synopsis

**Comment by BillyONeal** [comment]
Thanks for the feature!

**Pull request merged**